specs/lightclient/beacon-chain.md

```diff
@@ -343,88 +340,48 @@ def get_unslashed_participating_indices(state: BeaconState, flag_index: int, epo
     active_increments = get_total_active_balance(state) // increment
-    for index in get_eligible_validator_indices(state):
+    for index in unslashed_participating_indices:
         base_reward = get_base_reward(state, index)
-        if index in unslashed_participating_indices:
-            penalty_canceling_reward = base_reward * numerator // FLAG_DENOMINATOR
-            extra_reward = (
-                (base_reward * numerator * unslashed_participating_increments)
-                // (active_increments * FLAG_DENOMINATOR)
-            )
-            if is_in_inactivity_leak(state):
-                rewards[index] = penalty_canceling_reward
-            else:
-                rewards[index] = penalty_canceling_reward + extra_reward
+        # Cancel equivalent flag penalty
+        rewards[index] += Gwei(base_reward * numerator // FLAG_DENOMINATOR)
+        if not is_leak_active(state):
+            reward_numerator = base_reward * numerator * unslashed_participating_increments
+            rewards[index] += Gwei(reward_numerator // (active_increments * FLAG_DENOMINATOR))
     return rewards
 ```
 
-#### `get_regular_penalties`
+#### `get_flag_penalties`
 
 ```python
-def get_regular_penalties(state: BeaconState) -> Sequence[Gwei]:
-    """
-    Compute the regular epoch penalties for each validator.
-    """
+def get_flag_penalties(state: BeaconState) -> Sequence[Gwei]:
     penalties = [Gwei(0)] * len(state.validators)
-
-    # Only give penalties on period boundaries
-    if not is_activation_exit_period_boundary(state):
-        return penalties
-
-    attestation_numerators = [numerator for _, numerator in get_flag_indices_and_numerators()]
+    flag_numerators = [numerator for _, numerator in get_flag_indices_and_numerators()]
     for index in get_eligible_validator_indices(state):
-        # "Regular" penalty is applied to each eligible validator. If the validator participates in the attestation
-        # correctly, this penalty will be canceled.
-        penalty_per_epoch = sum(
-            get_base_reward(state, index) * numerator // FLAG_DENOMINATOR
-            for numerator in attestation_numerators
-        )
-        penalty_per_period = Gwei(penalty_per_epoch * EPOCHS_PER_ACTIVATION_EXIT_PERIOD)
-        penalties[index] = penalty_per_period
+        base_reward = get_base_reward(state, index)
+        penalty_per_epoch = sum(base_reward * numerator // FLAG_DENOMINATOR for numerator in flag_numerators)
+        # Cancel equivalent flag reward
+        penalties[index] += Gwei(penalty_per_epoch * EPOCHS_PER_ACTIVATION_EXIT_PERIOD)
     return penalties
 ```
 
 #### `get_leak_penalties`
 
 ```python
 def get_leak_penalties(state: BeaconState) -> Sequence[Gwei]:
-    """
-    Compute inactivity leak penalties for each validator.
-    """
     penalties = [Gwei(0)] * len(state.validators)
-
-    # Only give penalties on period boundaries
```

Comment:
The problem with removing this blocking condition is that test vectors can very strangely be generated. That is, the vectors for `get_leak_penalties` specifically if run not on an epoch boundary will return non-zero values. I'd prefer keeping the blocking condition in to avoid misuse in alternative contexts even though the calling function in the spec had the conditional